Refactor icon handling and improve error logging structure

peter9811 opened the pull request:
Enhance the icon handling logic for better clarity and add a fallback for unrecognized icons. Improve the error logging structure by separating critical and standard exception logging, and ensure more specific exception handling in the main application flow.

peter9811 committed:
Refactor icon handling and improve error logging structure

peter9811 committed:
Add error handling for access_handler import in cmdWorker.py

dependabot[bot] committed:
Bump jinja2 from 3.1.4 to 3.1.5

Bumps [jinja2](https://github.com/pallets/jinja) from 3.1.4 to 3.1.5.
- [Release notes](https://github.com/pallets/jinja/releases)
- [Changelog](https://github.com/pallets/jinja/blob/main/CHANGES.rst)
- [Commits](https://github.com/pallets/jinja/compare/3.1.4...3.1.5)

---
updated-dependencies:
- dependency-name: jinja2
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

p-yukusai commented on src/config.py:
```diff
@@ -79,168 +31,243 @@ def configFile():
+    document_folder = document_folder()
+    return os.path.exists(document_folder + PYWALL_INI)
+                    allValues = current.split(", ")
+                else:
```
What is that else doing there?

p-yukusai reviewed (changes requested):
Looks good, only issue is that random else in line 242 of src/config.py, it appears in the preview but not on the raw so that might be GitHub being silly.
Thanks for the PR, please get back to me when you can.

peter9811 committed:
Add package initialization files, update requirements, and enhance README and contributing guidelines

dependabot[bot] committed:
Bump the pip group across 1 directory with 3 updates

Bumps the pip group with 3 updates in the / directory: [jinja2](https://github.com/pallets/jinja), [requests](https://github.com/psf/requests) and [pillow](https://github.com/python-pillow/Pillow).


Updates `jinja2` from 3.1.4 to 3.1.5
- [Release notes](https://github.com/pallets/jinja/releases)
- [Changelog](https://github.com/pallets/jinja/blob/main/CHANGES.rst)
- [Commits](https://github.com/pallets/jinja/compare/3.1.4...3.1.5)

Updates `requests` from 2.30.0 to 2.32.2
- [Release notes](https://github.com/psf/requests/releases)
- [Changelog](https://github.com/psf/requests/blob/main/HISTORY.md)
- [Commits](https://github.com/psf/requests/compare/v2.30.0...v2.32.2)

Updates `pillow` from 9.5.0 to 10.3.0
- [Release notes](https://github.com/python-pillow/Pillow/releases)
- [Changelog](https://github.com/python-pillow/Pillow/blob/main/CHANGES.rst)
- [Commits](https://github.com/python-pillow/Pillow/compare/9.5.0...10.3.0)

---
updated-dependencies:
- dependency-name: jinja2
  dependency-type: indirect
  dependency-group: pip
- dependency-name: requests
  dependency-type: direct:production
  dependency-group: pip
- dependency-name: pillow
  dependency-type: direct:production
  dependency-group: pip
...

Signed-off-by: dependabot[bot] <support@github.com>

peter9811 committed:
Merge pull request #1 from peter9811/dependabot/pip/jinja2-3.1.5

Bump jinja2 from 3.1.4 to 3.1.5

peter9811 committed:
Merge pull request #2 from peter9811/dependabot/pip/pip-2ad13a3e10

Bump the pip group across 1 directory with 3 updates

peter9811 committed:
Implement core functionality for PyWall firewall management tool, including context menu installation, notification handling, and configuration management. Update requirements and add setup script for installation.

dependabot[bot] committed:
Bump requests from 2.32.2 to 2.32.3

Bumps [requests](https://github.com/psf/requests) from 2.32.2 to 2.32.3.
- [Release notes](https://github.com/psf/requests/releases)
- [Changelog](https://github.com/psf/requests/blob/main/HISTORY.md)
- [Commits](https://github.com/psf/requests/compare/v2.32.2...v2.32.3)

---
updated-dependencies:
- dependency-name: requests
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

p-yukusai commented:
I see you're doing a massive refactor which is great since the code was made ages ago when I was just learning how to program, please tell me when you're done to clone the whole thing to review and test it.

peter9811 commented:
@p-yukusai, probably is enough now. You can check it
If I find something else, I'll check it up on the future. 

Thanks for the app. I was here since 2022

peter9811 committed:
Add Prettier configuration and plugins for code formatting

peter9811 committed:
Fix formatting in LICENSE file for consistency in command representation

peter9811 committed:
Refactor code for improved readability and consistency in logging and path handling

dependabot[bot] committed:
Bump pyqt5 from 5.15.9 to 5.15.11

Bumps [pyqt5](https://www.riverbankcomputing.com/software/pyqt/) from 5.15.9 to 5.15.11.

---
updated-dependencies:
- dependency-name: pyqt5
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

peter9811 commented:
:)
Sorry, I got excited

peter9811 committed:
Merge pull request #7 from peter9811/dependabot/pip/requests-2.32.3

Bump requests from 2.32.2 to 2.32.3

peter9811 committed:
Merge pull request #4 from peter9811/dependabot/pip/pyqt5-5.15.11

Bump pyqt5 from 5.15.9 to 5.15.11

dependabot[bot] committed:
Bump psutil from 5.9.5 to 7.0.0

Bumps [psutil](https://github.com/giampaolo/psutil) from 5.9.5 to 7.0.0.
- [Changelog](https://github.com/giampaolo/psutil/blob/master/HISTORY.rst)
- [Commits](https://github.com/giampaolo/psutil/compare/release-5.9.5...release-7.0.0)

---
updated-dependencies:
- dependency-name: psutil
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump pillow from 10.3.0 to 11.1.0

Bumps [pillow](https://github.com/python-pillow/Pillow) from 10.3.0 to 11.1.0.
- [Release notes](https://github.com/python-pillow/Pillow/releases)
- [Changelog](https://github.com/python-pillow/Pillow/blob/main/CHANGES.rst)
- [Commits](https://github.com/python-pillow/Pillow/compare/10.3.0...11.1.0)

---
updated-dependencies:
- dependency-name: pillow
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump pytest from 7.3.1 to 8.3.4

Bumps [pytest](https://github.com/pytest-dev/pytest) from 7.3.1 to 8.3.4.
- [Release notes](https://github.com/pytest-dev/pytest/releases)
- [Changelog](https://github.com/pytest-dev/pytest/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pytest-dev/pytest/compare/7.3.1...8.3.4)

---
updated-dependencies:
- dependency-name: pytest
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

peter9811 committed:
Merge pull request #6 from peter9811/dependabot/pip/pillow-11.1.0

Bump pillow from 10.3.0 to 11.1.0

peter9811 committed:
Merge pull request #5 from peter9811/dependabot/pip/pytest-8.3.4

Bump pytest from 7.3.1 to 8.3.4

peter9811 committed:
Merge pull request #3 from peter9811/dependabot/pip/psutil-7.0.0

Bump psutil from 5.9.5 to 7.0.0

p-yukusai commented on README.md:
```diff
@@ -20,13 +20,50 @@
+## Features
+
+- Configure firewall rules
+- Monitor network traffic
```
PyWall doesn't monitor network traffic, is that a feature you're planning to add?

p-yukusai commented on README.md:
````diff
@@ -20,13 +20,50 @@
+```bash
+python main.py
+```
+
````
Installation should include directly downloading and running the exe from the releases, as that is what most users are accustomed to and what they usually do.

p-yukusai commented on main.py:
````diff
@@ -1,73 +1,198 @@
+
+if __name__ == "__main__":
+    main()
+```python
````
?

p-yukusai commented on main.py:
```diff
@@ -1,73 +1,198 @@
+        start(True)
+
+
+def main():
```
Two definitions for main?

p-yukusai commented on node_modules/.bin/prettier:
Node modules should be ignored

p-yukusai commented on node_modules/.bin/prettier.cmd:
Script should be in its own folder and not in node modules
e.g. $PyWall/scripts/script.suffix

p-yukusai commented on setup.py:
```diff
@@ -0,0 +1,37 @@
+
+setup(
+    name="PyWall",
+    version="1.0.0",
```
PyWall is not on version 1.0, please bump it later

p-yukusai commented on src/pop.py:
```diff
@@ -1,22 +1,104 @@
+    if getConfig("UI", "show_notifications") != "True":
+        return
+
+    if sys.platform == 'win32':
```
Unnecessary as PyWall is made exclusively for Windows

p-yukusai reviewed (changes requested):
I would tentatively approve the pull should the comments be resolved, pending a more manual review and testing after you finish committing changes.

peter9811 committed:
Update version to 0.9.0, add build and format scripts, and enhance README

peter9811 committed:
Refactor issue templates and CI configuration for consistency and clarity

dependabot[bot] committed:
Bump numpy from 1.24.3 to 2.0.2

Bumps [numpy](https://github.com/numpy/numpy) from 1.24.3 to 2.0.2.
- [Release notes](https://github.com/numpy/numpy/releases)
- [Changelog](https://github.com/numpy/numpy/blob/main/doc/RELEASE_WALKTHROUGH.rst)
- [Commits](https://github.com/numpy/numpy/compare/v1.24.3...v2.0.2)

---
updated-dependencies:
- dependency-name: numpy
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump pyqt5 from 5.15.10 to 5.15.11

Bumps [pyqt5](https://www.riverbankcomputing.com/software/pyqt/) from 5.15.10 to 5.15.11.

---
updated-dependencies:
- dependency-name: pyqt5
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

peter9811 committed:
Refactor imports and enhance GUI module loading with fallback mechanism

peter9811 committed:
Delete node_modules directory

peter9811 committed:
Merge pull request #9 from peter9811/dependabot/pip/pyqt5-5.15.11

Bump pyqt5 from 5.15.10 to 5.15.11

peter9811 committed:
Merge pull request #8 from peter9811/dependabot/pip/numpy-2.0.2

Bump numpy from 1.24.3 to 2.0.2

p-yukusai commented on README.md:
```diff
@@ -20,56 +20,99 @@
+## Features
+
+- Configure firewall rules
+- Block or allow connections based on IP, port, or application
```
Not true, we simply generate or delete a firewall rule for a specific file, we can generate different kinds of rules but they don't block based on IP or port but instead they block inbound or outbound connections per file using Window's firewall.

peter9811 replied:
Done

p-yukusai reviewed (changes requested):
Pedantic issue with the readme, only reason I'm requesting changes instead of changing it myself is to see if you had a reason to add this to the readme, other than that the code looks good to me, ill proceed to merge and test it when this last thing gets resolved.

peter9811 committed:
Update README.md

peter9811 committed:
Refactor and enhance configuration and logging modules

- Improved config handling with more robust methods in config.py
- Updated logger.py to use new logging utilities
- Enhanced configGui.py with better theme and config management
- Simplified imports and function names across multiple modules
- Added more robust error handling and logging mechanisms

peter9811 commented:
I did some edits on the error handling and try to do some changes on the contrast of the themes, but I don't want to change your style, so probably I'll let it to you

I was thinking on change the color of the text on the theme selector and maybe add some "shadow" or "glow", like bold style to make more readable, but...

Have a nice day

peter9811 commented:
I was having a hard time with a small "bug", when you try to add a word to the blacklist but the blacklist is empty, add a "," before the word, probably not critical, but can carry to maybe future bugs (idk)

p-yukusai commented:
I'll manually clone, review and create a new release when I have the time, but as is the code looks ready to merge, I will also look into the bug you talked about (based on the code and the comment I think you solved it, but I'll still check).
Piplock and maybe the pipfile itself needs an update, so the CLI stops complaining with PyQt's version, I'll do that as well.
Thank you for your time and your contributions!
